components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java

```diff
@@ -689,6 +689,31 @@ public APIDefinitionValidationResponse validateAPIDefinition(String apiDefinitio
         } else {
             Swagger swagger = parseAttemptForV2.getSwagger();
             Info info = swagger.getInfo();
+            // Validate the x-throttling-limit extension for API
+            String xThrottlingLimitExtension = APIConstants.SWAGGER_X_THROTTLING_LIMIT;
+            if (swagger.getVendorExtensions() != null && swagger.getVendorExtensions()
+                    .containsKey(xThrottlingLimitExtension)) {
+                OASParserUtil.validateAPIThrottleLimitExtension(validationResponse, swagger.getVendorExtensions()
+                                .get(xThrottlingLimitExtension));
+                if (validationResponse.getErrorItems().size() > 0) {
+                    return validationResponse;
+                }
+            }
+            // Validate the x-throttling-limit extension for operations
+            for (String path : swagger.getPaths().keySet()) {
+                Path pathItem = swagger.getPaths().get(path);
+                for (Operation operation: pathItem.getOperations()) {
+                    if (operation.getVendorExtensions() != null &&
+                            operation.getVendorExtensions().containsKey(xThrottlingLimitExtension)) {
+                        OASParserUtil.validateAPIThrottleLimitExtension(validationResponse,
+                                operation.getVendorExtensions().get(xThrottlingLimitExtension));
+                        if (validationResponse.getErrorItems().size() > 0) {
+                            return validationResponse;
```

Comment:
Modify the error item's description with more info like which path, which operation etc.